sympy/core/assumptions.py

```diff
@@ -610,55 +610,14 @@ def _ask(fact, obj):
     return None
 
 
-class ManagedProperties(type):
-    """Metaclass for classes with old-style assumptions"""
-    def __init__(cls, *args, **kws):
-        cls.__sympy__ = property(lambda self: True)
-
-        local_defs = {}
-        for k in _assume_defined:
-            attrname = as_property(k)
-            v = cls.__dict__.get(attrname, '')
-            if isinstance(v, (bool, int, type(None))):
-                if v is not None:
-                    v = bool(v)
-                local_defs[k] = v
-
-        defs = {}
-        for base in reversed(cls.__bases__):
-            assumptions = getattr(base, '_explicit_class_assumptions', None)
-            if assumptions is not None:
-                defs.update(assumptions)
-        defs.update(local_defs)
-
-        cls._explicit_class_assumptions = defs
-        cls.default_assumptions = StdFactKB(defs)
-
-        cls._prop_handler = {}
-        for k in _assume_defined:
-            eval_is_meth = getattr(cls, '_eval_is_%s' % k, None)
-            if eval_is_meth is not None:
-                cls._prop_handler[k] = eval_is_meth
-
-        # Put definite results directly into the class dict, for speed
-        for k, v in cls.default_assumptions.items():
-            setattr(cls, as_property(k), v)
-
-        # protection e.g. for Integer.is_even=F <- (Rational.is_integer=F)
-        derived_from_bases = set()
-        for base in cls.__bases__:
-            default_assumptions = getattr(base, 'default_assumptions', None)
-            # is an assumption-aware class
-            if default_assumptions is not None:
-                derived_from_bases.update(default_assumptions)
-
-        for fact in derived_from_bases - set(cls.default_assumptions):
-            pname = as_property(fact)
-            if pname not in cls.__dict__:
-                setattr(cls, pname, make_property(fact))
-
-        # Finally, add any missing automagic property (e.g. for Basic)
-        for fact in _assume_defined:
-            pname = as_property(fact)
-            if not hasattr(cls, pname):
-                setattr(cls, pname, make_property(fact))
+# XXX: ManagedProperties used to be the metaclass for Basic but now Basic does
+# not use a metaclass. We leave this here for backwards compatibility for now
+# in case someone has been using the ManagedProperties class in downstream
+# code. The reason that it might have been used is that when subclassing a
+# class and wanting to use a metaclass the metaclass must be a subclass of the
+# metaclass for the class that is being subclassed. Anyone wanting to subclass
+# Basic and use a metclass in their subclass would have needed to subclass
+# ManagedProperties. Here ManagedProperties is still the metaclass for Basic
+# but it is just type because that is the metaclass when metaclasses are not
+# being used.
+ManagedProperties = type
```

Comment:
Strictly speaking this makes it backwards incompatible as someone might be using this as the metaclass for something that is not also a `Basic`. Is there any way we can keep this class, but it gives a deprecation warning when initialized? Or would this cause too many problems with type comparisons?

Also a small typo: `and use a metclass`

Reply:
Most likely if anyone is using this it is because they are subclassing it and using it with a Basic subclass. Something like
```python
class MyMeta(ManagedProperties):
    # whatever MyMeta does

class MySubclass(Basic, metaclass=MyMeta):
    # whatever MySubclass does
```
The reason they might be using it here is just because they need to in order to be able to use `MyMeta` with `MySubclass` if it subclasses `Basic`. The question is whether any code like that can continue to work. If we leave the `ManagedProperties` class here then it won't be the subclass for Basic any more so then the code uses it will be mixing in something that is *not* the metaclass of `Basic`.

On the other hand it will still be a subclass of the metaclass (`type`) that *will* be used for `Basic` so maybe that can continue working although there might be some interaction between the `ManagedProperties.__init__` method and the `Basic.__init_subclass__` method. I guess I should add a test that a subclass set up in this way still works.

There is no reason why someone would need the `ManagedProperties.__init__` method after this change so `ManagedProperties` provides no more than `type`. It's possible though that they would have `MyMeta.__init__` pass args and kwargs to `ManagedProperties.__init__` which would fail if `ManagedProperties` is just an alias for `type`.

We could have the `__init__` (as the only funcional method of `ManagedProperties`) give a deprecation warning. That might make it difficult to write code that needs to straddle multiple SymPy versions though.

Reply:
The metaprogramming in the core is complicated enough without having to maintain backwards compatibility with it. The logic on ManagedProperties was never designed to be applied to anything other than Basic, and I'm doubtful it would even work.

Reply:
I've added ManagedProperties back with a warning.

Reply:
It's easy enough to test a basic level of compatibility. If someone is messing deep enough with metaclasses then it's probably impossible to maintain full compatibility but I think I've got it so that the most likely cases would be handled. Possibly no one is using ManagedProperties but we'll see.

Reply:
Fixed the metclass typo.